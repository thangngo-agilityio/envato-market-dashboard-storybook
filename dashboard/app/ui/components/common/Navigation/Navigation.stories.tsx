import type { Meta, StoryObj } from '@storybook/react';
import { Box, Flex, Text } from '@chakra-ui/react';

// Components
import Navigation from '.';

// Assets
import { DashboardIcon } from '@/ui/components/';

const meta: Meta<typeof Navigation> = {
  title: 'Custom Components/Navigation',
  tags: ['autodocs'],
  component: Navigation,
  argTypes: {
    children: {
      description:
        'The children of the navigation, e.g. text, react components, etc.',
    },

    destination: {
      description: 'The destination of the navigation component',
      defaultValue: '/',
    },
  },
  decorators: [
    (Story) => (
      <Box bg="background.component.primary">
        <Story />
      </Box>
    ),
  ],
  parameters: {
    controls: {
      expanded: true,
    },
  },
};

export default meta;

type Story = StoryObj<typeof Navigation>;

export const Mini: Story = {
  args: {
    destination: '/test',
    children: <DashboardIcon />,
  },
};

export const Expanded: Story = {
  args: {
    destination: '/test',
    children: (
      <Flex alignItems="center" justifyContent="space-between">
        <Flex gap={2.5} alignItems="center" justifyContent="space-between">
          <DashboardIcon />

          <Text
            fontSize="lg"
            fontWeight="medium"
            color="text.primary"
            transition="all .25s ease-in-out"
            fontFamily="primary"
          >
            Sample Navigation
          </Text>
        </Flex>
      </Flex>
    ),
  },
};
