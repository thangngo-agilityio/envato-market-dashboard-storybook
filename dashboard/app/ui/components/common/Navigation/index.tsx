import { MouseEvent, ReactNode, memo } from 'react';
import { Link } from '@chakra-ui/next-js';
import { usePathname } from 'next/navigation';
import isEqual from 'react-fast-compare';

export type TNavigationProps = {
  children: ReactNode;
  destination?: string;
  onClick?: (e: MouseEvent<HTMLAnchorElement>) => void;
};

const Navigation = ({
  children,
  destination = '/',
  onClick,
}: TNavigationProps) => {
  const pathname = usePathname();

  const isActive: boolean = `/${pathname?.slice(1)}` === destination;

  return (
    <Link
      href={destination}
      onClick={onClick}
      aria-label="navigate-item"
      variant={isActive ? 'primary' : 'default'}
    >
      {children}
    </Link>
  );
};

const NavigationComponent = memo(Navigation, isEqual);

export default NavigationComponent;
