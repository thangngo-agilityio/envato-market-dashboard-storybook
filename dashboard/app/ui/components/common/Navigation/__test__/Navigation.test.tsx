import { render } from '@testing-library/react';
import Navigation from '..';

describe('Navigation test case', () => {
  it('should render correctly with the default destination', () => {
    const { container } = render(<Navigation>Mock navigation</Navigation>);
    expect(container).toMatchSnapshot();
  });
});
