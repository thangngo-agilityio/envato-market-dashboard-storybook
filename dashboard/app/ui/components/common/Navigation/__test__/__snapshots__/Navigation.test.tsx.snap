// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation test case should render correctly with the default destination 1`] = `
<div>
  <a
    aria-label="navigate-item"
    class="chakra-link css-0"
    href="/"
  >
    Mock navigation
  </a>
</div>
`;
